python-backend/app/api/mines/mine/models/mine_type_detail.py

```diff
@@ -20,18 +20,28 @@ class MineTypeDetail(AuditMixin, Base):
         db.ForeignKey('mine_disturbance_code.mine_disturbance_code'),
         nullable=False
     )
+    active_ind = db.Column(
+        db.Boolean,
+        nullable=False,
+        default=True
+    )
 
 
     def __repr__(self):
-        return '<MineDisturbanceCode %r>' % self.mine_disturbance_code
+        return '<MineTypeDetail %r>' % self.mine_type_detail_xref_guid
 
     def json(self):
         return {
             'mine_type_detail_guid': str(self.mine_type_detail_xref_guid),
             'mine_type_guid': str(self.mine_type_guid),
-            'mine_disturbance_code': self.mine_disturbance_code
+            'mine_disturbance_code': self.mine_disturbance_code,
+            'active_ind': self.active_ind
         }
 
+    def validate_disturbance_code_with_tenure(self, mine_tenure_type_code):
+        assert mine_tenure_type_code in DISTURBANCE_CODES_CONFIG[self.mine_disturbance_code]['mine_tenure_type_codes']
```

Comment:
I agree that this isn't great. In discussions with @klefler, the decision was to put this constraint in the code at this point instead of at the database level. The issue is that the disturbances don't know anything about the tenure types under which they fall in the current database model.

The plan may be to move this logic to the database in the future. Kristin may be able to provide more perspective on this, but I believe the config option was to avoid a dramatic data model overhaul.

All of the other validation exists at the database level.

Reply:
What do you think about moving this check to the resource level? I don't think our ORM (SQL Alchemy) data model's should represent anything other than our database, and ideally should be generated code which is something I plan to look into. These aren't business model's and shouldn't have other rules defined in them. Jason/Jason/Gyan and I were having this discussion last week and we settled on the resource being a not-ideal, but better place for business rules and logic.

Reply:
@FreshGyan and I chatted a while back about using fatter models for managing business logic. So, that's why I moved the validation to the model.

Sounds like there may be two different interpretations of what the plan is for handling business logic in the application. Putting the logic in the model follows what I had heard the plan would be.

Reply:
I pretty strongly disagree, but let's have that discussion outside the context of this PR and come up with consensus on a pattern. Business logic in the data models will break down when resources us the same data models but want different business rules to apply. We can see this starting with special json transforms that are in the data models written specifically for one resources use of that model. It also fails when a business model, or what we send and receive from the API does not directly represent the data structure, which will likely happen over time.

Reply:
Sounds good. There are some larger backend patterns that need to be settled on for sure. Things like standardized error handling, record saving, application-layer validation, etc...

Happy to follow the pattern once I know what it is.